src/tests/ftest/util/test_utils_pool.py

```diff
@@ -173,6 +175,39 @@ def identifier(self):
             identifier = self.label.value
         return identifier
 
+    @fail_on(CommandFailure)
+    def _dmg_ready(self, subcommand):
+        """Check if it's ready to call dmg command.
+
+        Args:
+            subcommand (str): Method name in this class.
+
+        Raises:
+            CommandFailure: If the control method is API.
+
+        Returns:
+            bool: True if the control method is dmg and self.dmg is defined.
+                False otherwise.
+
+        """
+        if self.control_method.value == self.USE_API:
+            raise CommandFailure(
+                "Error: control method {} not supported for {}".format(
+                    self.control_method.value, subcommand))
+
+        if self.control_method.value == self.USE_DMG and self.dmg:
+            return True
+
+        if self.control_method.value == self.USE_DMG:
+            self.log.error("Error: Undefined dmg command")
+
+        else:
+            self.log.error(
+                "Error: Undefined control_method: %s",
+                self.control_method.value)
+
+        return False
```

Comment:
Simplification suggestion:
```suggestion
        if self.control_method.value == self.USE_DMG and not self.dmg:
            raise CommandFailure(
                "Error: Undefined dmg command object for {}".format(
                    subcommand))

        if self.control_method.value != self.USE_DMG:
            raise CommandFailure(
                "Error: control method {} not supported for {}".format(
                    self.control_method.value, subcommand))
```